cylc/flow/scheduler.py

```diff
@@ -1267,36 +1267,48 @@ def release_queued_tasks(self):
             and self.auto_restart_time is None
         ):
             pre_prep_tasks = self.pool.release_queued_tasks()
-            if not pre_prep_tasks:
-                # No tasks to submit.
-                return
 
-            # Start the job submission process.
-            self.is_updated = True
-            self.reset_inactivity_timer()
+        elif (
+            self.should_auto_restart_now()
+            and self.auto_restart_mode == AutoRestartMode.RESTART_NORMAL
+        ):
+            # Need to get preparing tasks to submit before auto restart
+            pre_prep_tasks = [
+                itask for itask in self.pool.get_tasks()
+                if itask.state(TASK_STATUS_PREPARING)
+            ]
 
-            self.task_job_mgr.task_remote_mgr.rsync_includes = (
-                self.config.get_validated_rsync_includes())
+        # Return, if no tasks to submit.
+        else:
+            return
+        if not pre_prep_tasks:
+            return
 
-            meth = LOG.debug
-            if self.options.reftest or self.options.genref:
-                meth = LOG.info
-            for itask in self.task_job_mgr.submit_task_jobs(
-                self.workflow,
-                pre_prep_tasks,
-                self.server.curve_auth,
-                self.server.client_pub_key_dir,
-                self.config.run_mode('simulation')
-            ):
-                if itask.flow_nums:
-                    flow = ','.join(str(i) for i in itask.flow_nums)
-                else:
-                    flow = FLOW_NONE
-                meth(
-                    f"{itask.identity} -triggered off "
-                    f"{itask.state.get_resolved_dependencies()}"
-                    f" in flow {flow}"
-                )
+        # Start the job submission process.
+        self.is_updated = True
+        self.reset_inactivity_timer()
+
+        self.task_job_mgr.task_remote_mgr.rsync_includes = (
+            self.config.get_validated_rsync_includes())
+
+        log = LOG.debug
+        if self.options.reftest or self.options.genref:
+            log = LOG.info
```

Comment:
```suggestion
log = LOG.info if (self.options.reftest or self.options.genref) else LOG.debug
```

Reply:
Doesn't fit on 1 line which is why I left it the way it is